cli/internal/reference/compref/compref.go

```diff
@@ -185,25 +185,70 @@ func Parse(input string) (*Ref, error) {
 		return nil, fmt.Errorf("invalid component name %q in %q, must match %q", ref.Component, originalInput, ComponentRegex)
 	}
 
-	// Step 6: Resolve type if not explicitly given
+	// Step 6: Build repository object using ParseRepository
+	var repositorySpec string
+	if ref.Type != "" {
+		repositorySpec = ref.Type + "::" + input
+	} else {
+		repositorySpec = input
+	}
+
+	repository, err := ParseRepository(repositorySpec)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse repository: %w", err)
+	}
+
+	ref.Repository = repository
+
+	// Extract the type from the parsed repository for consistency
 	if ref.Type == "" {
-		t, err := GuessType(input)
+		switch repository.(type) {
+		case *ociv1.Repository:
+			ref.Type = runtime.NewVersionedType(ociv1.Type, ociv1.Version).String()
+		case *ctfv1.Repository:
+			ref.Type = runtime.NewVersionedType(ctfv1.Type, ctfv1.Version).String()
+		}
+	}
+
+	return ref, nil
+}
+
+// ParseRepository parses a repository specification string and returns a typed repository object.
+// It accepts repository strings in the format:
+//   - [<type>::]<repository-spec>
+//
+// Where type can be "ctf" or "oci", and repository-spec is the actual repository location.
+// If no type is specified, it will be guessed using heuristics.
+func ParseRepository(repositorySpec string) (runtime.Typed, error) {
+	originalInput := repositorySpec
+	input := repositorySpec
+
+	// Extract optional type
+	var repoType string
+	if idx := strings.Index(input, "::"); idx != -1 {
```

Comment:
i find it very strange that you now reverse the string encoding that you just did in https://github.com/open-component-model/open-component-model/pull/628/files#diff-9b7244324b50840b90196d8c74c6124c37c9fcffedd5f45c2596cf116196d660R191. Cant you just call this method with a repo type directly?